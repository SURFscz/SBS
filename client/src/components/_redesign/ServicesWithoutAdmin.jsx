import React from "react";
import "./ServicesWithoutAdmin.scss";
import {stopEvent} from "../../utils/Utils";
import I18n from "../../locale/I18n";
import Entities from "./Entities";
import Logo from "./Logo";


class ServicesWithoutAdmin extends React.Component {

    openService = service => e => {
        if (e.metaKey || e.ctrlKey) {
            return;
        }
        stopEvent(e);
        this.props.history.push(`/services/${service.id}`);
    };

    render() {
        const {services} = this.props;
        const columns = [
            {
                nonSortable: true,
                key: "logo",
                header: "",
                mapper: service => <Logo src={service.logo}/>
            },
            {
                key: "name",
                header: I18n.t("models.services.name"),
                mapper: service => <a href={`/services/${service.id}`}
                                      className={"neutral-appearance"}
                                      onClick={this.openService(service)}>{service.name}</a>,
            },
            {
                key: "contact_email",
                header:  I18n.t("service.contact_email"),
                mapper: service => service.contact_email? service.contact_email : "-"
            }]
        return (
            <Entities entities={services}
                      modelName="servicesWithoutAdmin"
                      searchAttributes={["name"]}
                      defaultSort="name"
                      columns={columns}
                      rowLinkMapper={() => this.openService}
                      loading={false}
                      {...this.props}/>
        )
    }
}

export default ServicesWithoutAdmin;