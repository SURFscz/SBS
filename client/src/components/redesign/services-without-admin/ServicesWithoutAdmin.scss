table.servicesWithoutAdmin {

    thead {
        th {
            &.name {
                width: 40%;
                padding-left: 15px;
            }

            &.contactEmail {
                width: 25%;
            }
        }
    }

    tbody {
        td {
            &.name {
                padding-left: 15px;
            }

            &.service_memberships_count {
                text-align: center;
            }

            &.collaborations_count {
                text-align: center;
            }
        }
    }
}
