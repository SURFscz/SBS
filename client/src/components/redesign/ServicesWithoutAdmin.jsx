import React from "react";
import "./ServicesWithoutAdmin.scss";
import {stopEvent} from "../../utils/Utils";
import I18n from "i18n-js";
import Entities from "./Entities";
import Logo from "./Logo";


class ServicesWithoutAdmin extends React.Component {

    openService = service => e => {
        if (e.metaKey || e.ctrlKey) {
            return;
        }
        stopEvent(e);
        this.props.history.push(`/services/${service.id}`);
    };

    render() {
        const {services} = this.props;
        const columns = [
            {
                nonSortable: true,
                key: "logo",
                header: "",
                mapper: service => <Logo src={service.logo}/>
            },
            {
                key: "name",
                header: I18n.t("models.services.name"),
                mapper: service => <a href={`/services/${service.id}`} onClick={this.openService(service)}>{service.name}</a>,
            },
            {
                key: "contact_email",
                header:  I18n.t("service.contact_email"),
                mapper: service => service.contact_email? service.contact_email : "-"
            }]
        return (
            <Entities entities={services}
                      modelName="servicesWithoutAdmin"
                      searchAttributes={["name"]}
                      defaultSort="name"
                      columns={columns}
                      rowLinkMapper={() => this.openService}
                      loading={false}
                      {...this.props}/>
        )
    }
}

export default ServicesWithoutAdmin;