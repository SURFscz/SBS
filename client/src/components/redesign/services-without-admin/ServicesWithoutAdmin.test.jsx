import React from "react";
import { render } from "@testing-library/react";
import ServicesWithoutAdmin from "./ServicesWithoutAdmin";

describe("ServicesWithoutAdmin", () => {
    it("matches snapshot with default props", () => {
        const { asFragment } = render(
            <ServicesWithoutAdmin
                services={[]}
                onDeleteService={() => {}}
                onReassignService={() => {}}
                user={{ admin: false, service_memberships: [] }}
                entities={[]}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders with services", () => {
        const services = [
            { id: 1, name: "Service 1" },
            { id: 2, name: "Service 2" }
        ];
        const { asFragment } = render(
            <ServicesWithoutAdmin
                services={services}
                onDeleteService={() => {}}
                onReassignService={() => {}}
                user={{ admin: false, service_memberships: [] }}
                entities={[]}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
