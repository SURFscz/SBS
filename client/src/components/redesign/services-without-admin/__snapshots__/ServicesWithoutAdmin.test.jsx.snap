// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServicesWithoutAdmin matches snapshot with default props 1`] = `
<DocumentFragment>
  <div
    class="mod-entities "
  >
    <section
      class="entities-search"
    >
      <h2>
        Applications without administrators (0)
      </h2>
      <div
        class="search-filters servicesWithoutAdmin-search-filters"
      />
      <div
        class="search standalone"
      >
        <div
          class="sds--text-field sds--text-field--has-icon"
        >
          <div
            class="sds--text-field--shape"
          >
            <div
              class="sds--text-field--input-and-icon"
            >
              <input
                class="sds--text-field--input"
                placeholder="Search applications..."
                type="search"
                value=""
              />
              <span
                class="sds--text-field--icon"
              >
                <svg>
                  search.svg
                </svg>
              </span>
            </div>
          </div>
        </div>
      </div>
    </section>
    <section
      class="entities-list"
    />
    <div />
  </div>
</DocumentFragment>
`;

exports[`ServicesWithoutAdmin renders with services 1`] = `
<DocumentFragment>
  <div
    class="mod-entities "
  >
    <section
      class="entities-search"
    >
      <h2>
        Applications without administrators (0)
      </h2>
      <div
        class="search-filters servicesWithoutAdmin-search-filters"
      />
      <div
        class="search standalone"
      >
        <div
          class="sds--text-field sds--text-field--has-icon"
        >
          <div
            class="sds--text-field--shape"
          >
            <div
              class="sds--text-field--input-and-icon"
            >
              <input
                class="sds--text-field--input"
                placeholder="Search applications..."
                type="search"
                value=""
              />
              <span
                class="sds--text-field--icon"
              >
                <svg>
                  search.svg
                </svg>
              </span>
            </div>
          </div>
        </div>
      </div>
    </section>
    <section
      class="entities-list"
    />
    <div />
  </div>
</DocumentFragment>
`;
